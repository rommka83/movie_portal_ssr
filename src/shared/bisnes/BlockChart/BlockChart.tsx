import React, { FC, useState } from 'react';
import styles from './blockchart.module.css';

interface IProps {
  width: number;
  obj: {
    await: number;
    filmCritics: number;
    imdb: number;
    russianFilmCritics: number;
  };
}

export const BlockChart: FC<IProps> = ({ obj, width }) => {
  return (
    <div className={styles.root}>
      <input
        type='range'
        className={styles.inp}
        min='0'
        max='10'
        value={obj.await ?? 0}
        step='0,1'
        style={{ width: width }}
        readOnly
      />
      <input
        type='range'
        className={styles.inp}
        min='0'
        max='10'
        value={obj.filmCritics ?? 0}
        step='0,1'
        style={{ width: width }}
        readOnly
      />
      <input
        type='range'
        className={styles.inp}
        min='0'
        max='10'
        value={obj.imdb ?? 0}
        step='0,1'
        style={{ width: width }}
        readOnly
      />
      <input
        type='range'
        className={styles.inp}
        min='0'
        max='10'
        value={obj.russianFilmCritics ?? 0}
        step='0,1'
        style={{ width: width }}
        readOnly
      />
    </div>
  );
};
