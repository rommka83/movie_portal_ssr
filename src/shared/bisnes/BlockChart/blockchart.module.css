.root {
  display: flex;
  flex-direction: column;
}

.inp:not(:last-child) {
  margin-bottom: 3px;
}

.inp {
  height: 3px;
}
