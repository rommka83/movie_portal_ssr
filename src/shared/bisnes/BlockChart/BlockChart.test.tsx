import renderer from 'react-test-renderer';
import { BlockChart } from './BlockChart';

describe('проверка BlockChart', () => {
  test('отрисовка BlockChart ', () => {
    const component = renderer.create(
      <BlockChart
        width={0}
        obj={{
          await: 0,
          filmCritics: 0,
          imdb: 0,
          russianFilmCritics: 0,
        }}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
