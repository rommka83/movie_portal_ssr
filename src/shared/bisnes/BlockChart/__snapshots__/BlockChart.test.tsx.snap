// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка BlockChart отрисовка BlockChart  1`] = `
<div
  className="root"
>
  <input
    className="inp"
    max="10"
    min="0"
    readOnly={true}
    step="0,1"
    style={
      Object {
        "width": 0,
      }
    }
    type="range"
    value={0}
  />
  <input
    className="inp"
    max="10"
    min="0"
    readOnly={true}
    step="0,1"
    style={
      Object {
        "width": 0,
      }
    }
    type="range"
    value={0}
  />
  <input
    className="inp"
    max="10"
    min="0"
    readOnly={true}
    step="0,1"
    style={
      Object {
        "width": 0,
      }
    }
    type="range"
    value={0}
  />
  <input
    className="inp"
    max="10"
    min="0"
    readOnly={true}
    step="0,1"
    style={
      Object {
        "width": 0,
      }
    }
    type="range"
    value={0}
  />
</div>
`;
